CMakeLists.txt

```diff
@@ -738,70 +738,87 @@ if(WITH_OPTIM)
     add_definitions(-DWITH_OPTIM)
     if(BASEARCH_ARM_FOUND)
         add_definitions(-DARM_FEATURES)
-        if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-            if("${ARCH}" MATCHES "aarch64")
-                check_c_source_compiles(
-                    "#include <sys/auxv.h>
-                    int main() {
+        if("${ARCH}" MATCHES "aarch64")
+            check_c_source_compiles(
+                "#include <sys/auxv.h>
+                int main() {
+                    #if defined(__FreeBSD__) || defined(__OpenBSD__)
+                        unsigned long hwcap;
+                        elf_aux_info(AT_HWCAP, &hwcap, sizeof(hwcap));
+                        return (hwcap & HWCAP_CRC32);
+                    #else
                         return (getauxval(AT_HWCAP) & HWCAP_CRC32);
-                    }"
-                    ARM_AUXV_HAS_CRC32
-                )
-                if(ARM_AUXV_HAS_CRC32)
-                    add_definitions(-DARM_AUXV_HAS_CRC32)
-                else()
-                   message(STATUS "HWCAP_CRC32 not present in sys/auxv.h; cannot detect support at runtime.")
-                endif()
+                    #endif
+                }"
+                ARM_AUXV_HAS_CRC32
+            )
+            if(ARM_AUXV_HAS_CRC32)
+                add_definitions(-DARM_AUXV_HAS_CRC32)
+            else()
+               message(STATUS "HWCAP_CRC32 not present in sys/auxv.h; cannot detect support at runtime.")
+            endif()
+        else()
+            check_c_source_compiles(
+                "#include <sys/auxv.h>
+                int main() {
+                    #if defined(__FreeBSD__) || defined(__OpenBSD__)
+                        unsigned long hwcap2;
+                        elf_aux_info(AT_HWCAP2, &hwcap2, sizeof(hwcap2));
+                        return (hwcap2 & HWCAP2_CRC32);
+                    #else
+                        return (getauxval(AT_HWCAP2) & HWCAP2_CRC32);
+                    #endif
+                }"
+                ARM_AUXV_HAS_CRC32
+            )
+            if(ARM_AUXV_HAS_CRC32)
+                add_definitions(-DARM_AUXV_HAS_CRC32)
```

Comment:
A lot of unnecessary white space changes... Makes reviewing the code really hard.